### DISCUSSION TITLE
Add Dogecoin Address to UACOMMENT

### DISCUSSION THREAD

**Discussion by Much-Mining** [dogecoin/dogecoin#3019] [General]
There has been discussion about custom doge nodes on the tweeter about UACOMMENT & to add your name on your doge node so you can personalize it. 

This would easily allow for custom nodes to be tracked (if wanted) for tipping purposes & would allow the everyday computer user to easily add their own custom name. This I feel, would attract more node runners. The more the merrier! 

Obviously, this isn't as important as non-glitchy dog money, but it would be super cool!

**Comment by patricklodder** [comment]
Hello,

My name is Alice, if you read this it means my message through a series of mules made it out of the dungeons of the central processing facility of the State of Mars where I am currently detained, awaiting my sentenced recycling. I hope that by sharing my story with you, I can help you not make the same mistakes I did. 

You see, I was running a Dogecoin node with the awesome UAComment feature enabled, and I filled in my alias. Unfortunately, less than 24 hours ago, the State of Mars passed a law that per immediate outlawed all forms of financial transactions with nations in the Saturn sphere, because of their threats of invasion.

Unfortunately, Dogecoin has been designed to withstand censorship and allow as many incoming connections as possible, so I failed  to stop my node from relaying a transaction that sent DOGE out to an address belonging to the Warlord of Titan and it unfortunately relayed it straight to a government probe. It took under 6.9 seconds for the Mars Secret Police to teleport into my cabin and arrest me. 420 seconds later, I was sentenced to recycling.

Please, don't make my mistake, and be careful with what personal information you expose.

Alice

**Reply by Much-Mining** [reply]
Dear Alice: 

I am sorry to hear this has happened. We the "dogecoin community" take security and privacy seriously. When setting up your custom alias, there are not one but two accept confirmations / warnings about the risk being taken. The custom alias is for advanced node runners (who want to support the dogecoin network), which is why upon setup it does a confirmation check that port 22556 is indeed open before being applied. If no port is open, it disallows the custom alias. 

If open (which yours was), the second warning triggers and states that with an open port/custom alias you should refrain from using the Dogecoin Core as your wallet as it opens you to attacks. Having an offline Cold Storage device (as we have always suggested to the people of mars) would have protected the your Dogecoin from the Warlord of Titan. I understand the need for hot wallets and there are many alternative options available. However!

I get that some would want to use the "core" wallet anyway. I have suggested that an option to disable the use of the Dogecoin Node as a wallet if the port is open unless manually enabled in the config file. (Kind of like the alias is now) This will help protect end users from having the same predicament. I know this wasn't an option with you so don't take this personally, but if it were implemented, and you lost your Dogecoin, that's no fault but your own.

TLDR: If you enter a custom alias, warn multiple times you shouldn't use the node for dogecoin (or do so at your own risk anyway ) and verify ports open (which most end users can't do). If they want/can open the ports, they can assume the risk IMO.

**Reply by patricklodder** [reply]
> Having an offline Cold Storage device (as we have always suggested to the people of mars) would have protected the your Dogecoin from the Warlord of Titan

It was relay, not transacted from a local wallet. Relevance: A probe into limiting non-KYC tx relay was drafted by some European Union legislators earlier this year.

**Comment by patricklodder** [comment]
Dear community,

We need to stand up for our rights. We all enabled the list address feature on our Dogecoin Core nodes, but no one told us that there are all these [weaknesses](https://blog.trailofbits.com/2020/06/11/ecdsa-handle-with-care/) in the ECDSA algorithm that signs our transactions. And no one told us that we should read all the best practices about not reusing the same address, that [even Satoshi spoke of in 2008 in the original Bitcoin whitepaper](http://bitcoin.org/bitcoin.pdf).

Collectively, we have now lost 69B DOGE to hackers. This has to stop. I suggest we go and seek out all the dogecoin developers and shoot them. After all, they have built us this feature.

Regards, Bob.
